code/cubical/Grammar/Inductive/HLevels.agda

```diff
@@ -0,0 +1,163 @@
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.HLevels
+
+module Grammar.Inductive.HLevels (Alphabet : hSet ℓ-zero)where
+
+open import Cubical.Data.Empty
+open import Cubical.Data.Sigma
+open import Cubical.Data.Sum
+open import Cubical.Data.Unit
+open import Cubical.Data.W.Indexed
+
+open import Helper
+open import Grammar.Base Alphabet
+open import Grammar.HLevels Alphabet
+open import Grammar.Dependent Alphabet
+open import Grammar.Epsilon Alphabet
+open import Grammar.LinearProduct Alphabet
+open import Grammar.Lift Alphabet
+open import Grammar.Inductive.Indexed Alphabet as Inductive
+open import Term.Base Alphabet
+
+private
+  variable ℓG ℓG' ℓ ℓ' ℓ'' ℓ''' : Level
+
+isSetValued : ∀ {A : Type ℓ} → Functor A → Type ℓ
+isSetValued (k g) = isSetGrammar g
+isSetValued {A = A} (Var a) = Unit*
```

Comment:
I'm a bit confused how you can get away with unit here. At first glance it seems like you're assuming that all of the recursive calls are set, but there's no way that works so I think I'm missing something

Reply:
Is it accurate to say that the proofs that each recursive call is a set is handled on line 119, where you require the user to pass an argument that the grammar being plugged into the endofunctor is a set?


```
(g : A → SetGrammar ℓ')
...
isSet⟦F⟧ (Var a) isSetF g = isSetGrammarLift (g a .snd)
```